hook.js

```diff
@@ -116,6 +116,24 @@ function isBareSpecifier (specifier) {
   }
 }
 
+function mapExcludingDuplicates () {
+  const map = new Map()
+  const duplicates = new Set()
+  return {
+    set (k, v) {
+      if (map.has(k)) {
+        duplicates.add(k)
+        map.delete(k)
+      } else {
```

Comment:
I think this might have to be a `else if (!duplicates.has(k))`, otherwise on the next run it can get added again.